### ISSUE TITLE
`libusb_set_interface_alt_setting` unlocks device handle lock without having it locked first if device has been disconnected

### ISSUE THREAD

**Issue by papr** [libusb/libusb#1199]
Dear maintainers,

TLDR: I believe [this change](https://github.com/libusb/libusb/commit/1a08aa84d96397a3840a75abe66051f5360c2c84#diff-c1f9bc250077d41456a3e580fca0ddf5d8c25b741bff6d9b9505990a8b70b254L1766-L1772) has introduced a regression that causes [`usbi_mutex_destroy(&dev_handle->lock)`](https://github.com/libusb/libusb/blob/master/libusb/core.c#L1440) to fail with EBUSY if `libusb_set_interface_alt_setting` and `libusb_close` have been called after the device has been disconnected.

Context: I am a libuvc user (libuvc linking against the current libusb main) who
1. calls [`uvc_close`](https://github.com/libuvc/libuvc/blob/master/src/device.c#L1721-L1749) once the camera has failed to provide frames for a while, e.g. due to a disconnect.
2. `uvc_close` will attempt to release the interface which will call [`libusb_set_interface_alt_setting`](https://github.com/libuvc/libuvc/blob/master/src/device.c#L1023).
3. If the device had been disconnected, [this code block will be executed](https://github.com/libusb/libusb/blob/master/libusb/core.c#L1765-L1768), unlocking the `dev_handle->lock` without having it locked first.
4. Later, in `libusb_close`/`do_close`, [`usbi_mutex_destroy(&dev_handle->lock)`](https://github.com/libusb/libusb/blob/master/libusb/core.c#L1440) will fail because the lock is in an unexpected state

https://github.com/libusb/libusb/commit/1a08aa84d96397a3840a75abe66051f5360c2c84#diff-c1f9bc250077d41456a3e580fca0ddf5d8c25b741bff6d9b9505990a8b70b254L1766-L1772 moved the `usbi_mutex_lock` call without removing the corresponding `usbi_mutex_unlock(&dev_handle->lock)` call.

**Comment by mcuee** [comment]
Fixed by #1200.
